### PR TITLE
Fetch csv from s3 bucket

### PR DESCRIPTION
**Pull request by DharmikGadhiya**
New PR for, https://github.com/fivetran/functions/pull/3

### CODE REVIEW

File: file/index_csv.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+const AWS = require('aws-sdk');
+let s3 = new AWS.S3();
+
+let bucketName = 'lambda-function-records';
+let bucketParams = {
+  Bucket: bucketName
+};
+
+let limitNumberOfFileToUpdateInOneRun = 2;
```
Comment (tabalinas) [inline file/index_csv.js]: ...NumberOfFiles..

File: file/index_csv.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+};
+
+async function update(state, secrets, callback) {
+  let response = initializeResponse(state);
+
+  let modifiedFiles = [];
+
+  let listObjectsPromise = s3.listObjects(bucketParams).promise();
+
+  listObjectsPromise.then((result) => {
```
Comment (tabalinas) [inline file/index_csv.js]: `await s3.listObjects(bucketParams).promise().then...`

File: file/index_csv.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+      Key: modifiedFile.Key
+    };
+
+    await s3.getObject(params).promise().then((result) => {
+      let fileData = result.Body.toString('utf-8');
+      let rows = fileData.split("\n");
+      let headers;
+      let count = 0;
+
+      for (let i in rows) {
```
Comment (tabalinas) [inline file/index_csv.js]: don't use for-in to iterate over arrays

File: file/index_csv.js [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        count++;
+        let cols = rows[i].split(",");
+        // Extract headers
+        if (count === 1) {
+          headers = cols;
+          continue;
+        }
+
+        let obj = {};
+        for (let j in cols) {
```
Comment (tabalinas) [inline file/index_csv.js]: same here

File: file/index_csv.js [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+  // Waiting for listing of objects so we can get all modified files
+  await listObjectsPromise;
+
+  // Sort in ascending order
+  modifiedFiles.sort((a, b) => {
+    return Date.parse(a.LastModified) - Date.parse(b.LastModified);
+  });
+
+  // Process files one by one
+  for (let modifiedFile of modifiedFiles) {
```
Comment (tabalinas) [inline file/index_csv.js]: and here

Comment (tabalinas) [inline file/index_csv.js]: on prev review I proposed using `forEach`, seems to be the best option

**Comment by DharmikGadhiya** [comment]
@tabalinas When I used for-each it was giving some errors, for I kept it as it is, Other comments are addressed

File: file/csv/aws_lambda/index.js [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+  // Process files one by one
+  for (let modifiedFile of modifiedFiles) {
+
+    let params = {
+      Bucket: bucketName,
+      Key: modifiedFile.Key
+    };
+
+    await s3.getObject(params).promise().then((result) => {
+      let fileData = result.Body.toString('utf-8');
```
Comment (tabalinas) [inline file/csv/aws_lambda/index.js]: be consistent about quotes: use double quotes everywhere

**Pull request merged**